pyOCD/target/cortex_m.py

```diff
@@ -488,7 +488,10 @@ def writeMemory(self, addr, value, transfer_size = 32):
         write a memory location.
         By default the transfer size is a word
         """
-        self.transport.writeMem(addr, value, transfer_size)
+        try:
+            self.transport.writeMem(addr, value, transfer_size)
+        except:
+            self.handleMemoryError()
```

Comment:
As is, the code calling writeMemory has no way to tell that the function failed.  It might be good to re-raise the exception, or raise a another.  That way the calling code would know the write failed, and that it should retry the write, or sequence of writes.